src/citrine/resources/data_concepts.py

```diff
@@ -291,6 +291,11 @@ def _build_list_of_soft_links(dc_obj, obj_with_soft_links, field: str, reverse_f
         if linked_objects is None:
             return
 
+        # Make `linked_objects` point to the list of linked objects themselves, as opposed to the
+        # attribute of `obj_with_soft_links`. The list in `obj_with_soft_links` will be modified
+        # in the loop below, and we don't want to modify a list while iterating over it.
+        linked_objects = [l for l in linked_objects]
```

Comment:
Is there a reason to copy with this list comprehension instead of python's [copy module](https://docs.python.org/2/library/copy.html) (as used [in taurus](https://github.com/CitrineInformatics/taurus/blob/develop/taurus/util/impl.py#L113))?

Reply:
No reason not to use `copy` (which I think is equivalent to this list comprehension). I changed it to `copy` for readability.

`deepcopy` does not work, because it is important that `setattr` below act on the actual objects that `obj_with_soft_links` points to, not copies of those objects.